brave/src/test/java/brave/features/handler/MutableSpanAsyncReporterTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.features.handler;
+
+import brave.Tags;
+import brave.Tracing;
+import brave.handler.MutableSpan;
+import brave.handler.MutableSpanBytesEncoder;
+import brave.handler.SpanHandler;
+import brave.propagation.B3SingleFormat;
+import brave.propagation.TraceContext;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import zipkin2.codec.BytesEncoder;
+import zipkin2.codec.Encoding;
+import zipkin2.junit.ZipkinRule;
+import zipkin2.reporter.AsyncReporter;
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This is an example of why {@link MutableSpanBytesEncoder} was written. Particularly, it allows
+ * direct encoding from {@link MutableSpan} into JSON without converting to Zipkin model first.
+ */
+public class MutableSpanAsyncReporterTest {
```

Comment:
here's the cool file. Let's us plug in as an encoder similar to stackdriver https://github.com/openzipkin/zipkin-gcp/tree/master/sender-stackdriver#configuration